pkg/extension/sumologicextension/extension.go

```diff
@@ -34,30 +35,39 @@ type sumologicExtension struct {
 	conf             *Config
 	logger           *zap.Logger
 	registrationInfo OpenRegisterResponsePayload
+	closeChan        chan struct{}
 }
 
 const (
 	// TODO: fix
-	niteBaseUrl = "https://nite-open-events.sumologic.net/"
+	niteBaseUrl              = "https://nite-open-events.sumologic.net/"
+	heartbeatUrl             = "api/v1/collector/heartbeat"
+	registerUrl              = "api/v1/collector/register"
+	defaultHeartbeatInterval = 15 * time.Second
 )
 
 func newSumologicExtension(conf *Config, logger *zap.Logger) (*sumologicExtension, error) {
 	if conf.CollectorName == "" {
 		return nil, errors.New("collector name is unset")
 	}
+	if conf.HeartBeatInterval <= 0 {
+		conf.HeartBeatInterval = defaultHeartbeatInterval
+	}
 
 	return &sumologicExtension{
 		// TODO: don't hardcode
-		baseUrl: niteBaseUrl,
-		conf:    conf,
-		logger:  logger,
+		baseUrl:   niteBaseUrl,
+		conf:      conf,
+		logger:    logger,
+		closeChan: make(chan struct{}),
 	}, nil
 }
 
 func (pm *sumologicExtension) Start(ctx context.Context, host component.Host) error {
 	if err := pm.register(ctx, pm.conf.CollectorName); err != nil {
 		return err
 	}
+	go pm.heartbeat(ctx)
```

Comment:
That ain't right: the `ctx` will outlive the `Start`'s scope. I'd drop this param, see my other comment